### PR TITLE
Authentication Config Examples

### PR DESCRIPTION
**Pull request by EngHabu**
Signed-off-by: Haytham Abuelfutuh <haytham@afutuh.com>

Update documentation with steps on how to configure auth for a few popular services...

#657
#925

### CODE REVIEW

File: rsts/howto/authentication/setup.rst [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+4. OPTIONAL: Add logout redirect URIs (e.g. http://localhost:30081/logout for sandbox)
+5. Note down the Client ID and Client Secret
+
+KeyCloak OpenID Connect
+------------------------
+
+`KeyCloak <https://www.keycloak.org/>`__ is an open source solution for authentication, it supports both OpenID Connect and OAuth2 protocols (among others). 
+KeyCloak can be configured to be both the OpenID Connect and OAuth2 Authorization Server provider for flyte.
+
+<TBD>
```
Comment (kumare3) [inline rsts/howto/authentication/setup.rst:58]: lets add a 
.. caution:: Coming soon!?

File: rsts/howto/authentication/setup.rst [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+The following is a listing of the Flytekit configuration values we set in CI, along with a brief explanation.
+
+* ``FLYTE_CREDENTIALS_CLIENT_ID`` and ``FLYTE_CREDENTIALS_CLIENT_SECRET``
+  When using basic authentication, this is the username and password.
+* ``export FLYTE_CREDENTIALS_AUTH_MODE=basic``
+  This tells the SDK to use basic authentication. If not set, Flytekit will assume you want to use the standard OAuth based three-legged flow.
+* ``export FLYTE_CREDENTIALS_AUTHORIZATION_METADATA_KEY=text``
+  At Lyft, we set this to conform to this `header config <https://github.com/flyteorg/flyteadmin/blob/eaca2fb0e6018a2e261e9e2da8998906477cadb5/pkg/auth/config/config.go#L53>`_ on the Admin side.
```
Comment (kumare3) [inline rsts/howto/authentication/setup.rst:253]:
```suggestion
  At Lyft, the value is set to conform to this `header config <https://github.com/flyteorg/flyteadmin/blob/eaca2fb0e6018a2e261e9e2da8998906477cadb5/pkg/auth/config/config.go#L53>`_ on the Admin side.
```

**Comment by SandraGH5** [comment]
Thumbs up!

File: rsts/howto/authentication/setup.rst [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+`KeyCloak <https://www.keycloak.org/>`__ is an open source solution for authentication, it supports both OpenID Connect and OAuth2 protocols (among others). 
+KeyCloak can be configured to be both the OpenID Connect and OAuth2 Authorization Server provider for flyte.
+
+.. caution:: Coming soon!
+
+Apply configuration
+-------------------
+
+1. Store the `client_secret` in a k8s secrt as follows:
```
Comment (katrogan) [inline rsts/howto/authentication/setup.rst:63]:
```suggestion
1. Store the `client_secret` in a k8s secret as follows:
```

File: rsts/howto/authentication/setup.rst [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+        userAuth:
+            openId:
+                baseUrl: https://dev-14186422.okta.com/oauth2/auskngnn7uBViQq6b5d6 # Okta with a custom Authorization Server
+                scopes:
+                - profile
+                - openid
+                # - offline_access # Uncomment if OIdC supports issuing refresh tokens.
+                clientId: 0oakkheteNjCMERst5d6
+
+1. Store flyte propeller's `client_secret` in a k8s secrt as follows:
```
Comment (katrogan) [inline rsts/howto/authentication/setup.rst:191]:
```suggestion
1. Store flyte propeller's `client_secret` in a k8s secret as follows:
```

**Review by katrogan — APPROVED** [review]
this is amazingly documented!

**Review by katrogan — APPROVED** [review]

**Review by wild-endeavor — APPROVED** [review]

**Pull request merged**